src/solver/solver.cxx

```diff
@@ -1226,51 +1221,37 @@ Field3D Solver::globalIndex(int localStart) {
  * Running user-supplied functions
  **************************************************************************/
 
-void Solver::setSplitOperator(rhsfunc fC, rhsfunc fD) {
-  split_operator = true;
-  phys_conv = fC;
-  phys_diff = fD;
-}
-
 int Solver::run_rhs(BoutReal t) {
   int status;
-  
+
   Timer timer("rhs");
-  
-  if(split_operator) {
+
+  if (model->splitOperator()) {
     // Run both parts
-    
+
     int nv = getLocalN();
     // Create two temporary arrays for system state
     Array<BoutReal> tmp(nv);
     Array<BoutReal> tmp2(nv);
-    
+
     save_vars(tmp.begin()); // Copy variables into tmp
     pre_rhs(t);
-    if(model) {
-      status = model->runConvective(t);
-    }else 
-      status = (*phys_conv)(t);
+    status = model->runConvective(t);
     post_rhs(t); // Check variables, apply boundary conditions
-    
-    load_vars(tmp.begin()); // Reset variables
+
+    load_vars(tmp.begin());   // Reset variables
     save_derivs(tmp.begin()); // Save time derivatives
     pre_rhs(t);
-    if(model) {
-      status = model->runDiffusive(t, false);
-    }else
-      status = (*phys_diff)(t);
+    status = model->runDiffusive(t, false);
     post_rhs(t);
     save_derivs(tmp2.begin()); // Save time derivatives
-    for(BoutReal *t = tmp.begin(), *t2 = tmp2.begin(); t != tmp.end(); ++t, ++t2)
+    for (BoutReal *t = tmp.begin(), *t2 = tmp2.begin(); t != tmp.end(); ++t, ++t2) {
```

Comment:
I think this would shut up clang-tidy, but it still involves pointer arithmetic behind the scenes, because `Array::begin` returns a pointer:

```cpp
std::transform(tmp.begin(), tmp.end(), tmp2.begin(), tmp.begin(),
               [](BoutReal& t1, BoutReal t2) { return t1 += t2; });
```